### PR TITLE
Move to struct init intention: missed some parens-related cases

### PR DESCRIPTION
**Pull request by wbars**

### CODE REVIEW

File: src/com/goide/intentions/GoMoveToStructInitializationIntention.java [hunk 1]
```diff
@@ -116,18 +116,24 @@ private static boolean isFieldDefinition(@Nullable PsiElement element) {
...
 
     PsiElement resolve = rightExpression != null ? rightExpression.resolve() : null;
     GoStatement previousElement = resolve != null ? PsiTreeUtil.getPrevSiblingOfType(assignment, GoStatement.class) : null;
     return previousElement != null && exists(getLeftHandElements(previousElement), e -> isResolvedTo(e, resolve));
   }
 
+  @NotNull
+  private static GoExpression getAnchor(@NotNull GoExpression expression) {
+    GoExpression parentExpression = PsiTreeUtil.getTopmostParentOfType(expression, GoExpression.class);
+    return parentExpression != null ? parentExpression : expression;
```
Comment (zolotov) [inline src/com/goide/intentions/GoMoveToStructInitializationIntention.java]: ObjectUtils#notNull

File: src/com/goide/intentions/GoMoveToStructInitializationIntention.java [hunk 2]
```diff
@@ -116,18 +116,24 @@ private static boolean isFieldDefinition(@Nullable PsiElement element) {
...
+    GoReferenceExpression rightExpression =
+      unwrapParensAndCast(GoPsiImplUtil.getRightExpression(assignment, getAnchor(referenceExpression)));
 
     PsiElement resolve = rightExpression != null ? rightExpression.resolve() : null;
     GoStatement previousElement = resolve != null ? PsiTreeUtil.getPrevSiblingOfType(assignment, GoStatement.class) : null;
     return previousElement != null && exists(getLeftHandElements(previousElement), e -> isResolvedTo(e, resolve));
   }
 
+  @NotNull
+  private static GoExpression getAnchor(@NotNull GoExpression expression) {
```
Comment (zolotov) [inline src/com/goide/intentions/GoMoveToStructInitializationIntention.java]: name could be more descriptive

**Comment by wbars** [comment]
Force updated

**Comment by grenki** [comment]
I think it's all right.

**Comment by zolotov** [comment]
Thank you. Merged

**Pull request merged**